mlflow/pipelines/regression/v1/pipeline.py

```diff
@@ -579,18 +542,69 @@ def read_dataframe(artifact_name, output_dir, file_name, step_name):
                 return None
 
         elif artifact_name == "ingested_scoring_data":
-            return read_dataframe(
-                "ingested_scoring_data",
-                ingest_scoring_output_dir,
-                IngestScoringStep._DATASET_OUTPUT_NAME,
-                ingest_scoring_step.name,
-            )
+            return read_dataframe_from_path(artifact_path, ingest_scoring_step.name)
 
         elif artifact_name == "scored_data":
-            return read_dataframe(
-                "scored_data", predict_output_dir, _SCORED_OUTPUT_FILE_NAME, predict_step.name
+            return read_dataframe_from_path(artifact_path, predict_step.name)
+
+        else:
+            raise MlflowException(
+                f"The artifact with name '{artifact_name}' is not supported.",
+                error_code=INVALID_PARAMETER_VALUE,
             )
 
+    def _get_artifact_path(self, artifact_name: str) -> Optional[str]:
+        """
+        Returns a path that may or may not exist
+        """
+        (
+            ingest_scoring_step,
+            predict_step,
+            ingest_step,
+            split_step,
+            transform_step,
+            train_step,
+            _,
+            register_step,
+        ) = self._steps
+
+        ingest_output_dir = get_step_output_path(self._pipeline_root_path, ingest_step.name, "")
+        split_output_dir = get_step_output_path(self._pipeline_root_path, split_step.name, "")
+        transform_output_dir = get_step_output_path(
+            self._pipeline_root_path, transform_step.name, ""
+        )
+        train_output_dir = get_step_output_path(self._pipeline_root_path, train_step.name, "")
+
+        ingest_scoring_output_dir = get_step_output_path(
+            self._pipeline_root_path, ingest_scoring_step.name, ""
+        )
+        predict_output_dir = get_step_output_path(self._pipeline_root_path, predict_step.name, "")
+        register_output_dir = get_step_output_path(self._pipeline_root_path, register_step.name, "")
```

Comment:
Shall we move these into the if-else clause so that only the directory needed by the artifact is computed? The cost of doing so is low in current implementation but it could be expensive if we separate the output storage to some remote location.